app/controllers/api/widgets_controller.rb

```diff
@@ -12,8 +12,8 @@ def generate_content_resource(type, id, _data = nil)
 
     def generate_widget_content(widget)
       desc = "#{widget_ident(widget)} content generation"
-      task_id = queue_object_action(widget, desc, :method_name => "generate_content", :args => ["User", User.current_user.current_group.description, [User.current_user.userid]])
-      action_result(true, desc, :task_id => task_id)
+      widget.queue_generate_content
+      action_result(true, desc)
```

Comment:
@d-m-u 
`widget.queue_generate_content` returns task_id

so can we add task id to result as well ?

  `action_result(true, desc, :task_id => task_id)`

Reply:
No, it doesn't. It will for a user. It does not for a group.